src/components/Modal/Modal.js

```diff
@@ -29,30 +29,47 @@ const Modal = ({ showModal, handleModalClose, deleteItem, item }) => {
         }
       }
     };
+
+    const closeOnClickOutside = (e) => {
```

Comment:
Good thinking on this, I think this is a really important feature to have. I think this is almost always how I close modals.